### PR TITLE
Fix the exception when export saving order is less than 3 in export tab of preferences.

### LINKED ISSUE #10157: export sort order: use specified order
### JabRef version

5.9 (latest release)

### Operating system

Windows

### Details on version and operating system

Windows 10 22H2

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Go to preferences
2. Go to Import and export
3. Select Use specified order
4. Remove the last sort order by using the "-" button
5. Click Save
6. Exception occurs

afbeelding

### Appendix

...
<details>

<summary>Log File</summary>

```
Paste an excerpt of your log file here
```

</details>

### PR DESCRIPTION
**Pull request by papatekken**
When saving the Preferences, if  no. of items in export tab->saving order list is less than 3, exception is raised.

So in this change, it will check the no. of items in saving order list before save to corresponding prerferences attribute.

That will fix #10157

### Mandatory checks
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -349,6 +349,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 - We fixed the shared database opening dialog to remember autosave folder and tick. [#7516](https://github.com/JabRef/jabref/issues/7516)
 - We fixed an issue where name formatter could not be saved. [#9120](https://github.com/JabRef/jabref/issues/9120)
 - We fixed a bug where after the export of Preferences, custom exports were duplicated. [#10176](https://github.com/JabRef/jabref/issues/10176)
+- we fixed a bug where exception was raised when saving less than 3 export save order in preference. [#10157](https://github.com/JabRef/jabref/issues/10157)
```
Comment (koppor) [inline CHANGELOG.md:352]: Quick comment: Please move this one up to the Unreleased part, because we released JabRef 5.10 meanwhile.

**Review by Siedlerchr — APPROVED** [review]
Thanks for your contribution! Overall looks good, just the changelog and then it's good from my side

File: src/main/java/org/jabref/preferences/JabRefPreferences.java [hunk 2]
```diff
@@ -2253,12 +2259,28 @@ private void storeExportSaveOrder(SaveOrder saveOrder) {
         putBoolean(EXPORT_IN_ORIGINAL_ORDER, saveOrder.getOrderType() == SaveOrder.OrderType.ORIGINAL);
         putBoolean(EXPORT_IN_SPECIFIED_ORDER, saveOrder.getOrderType() == SaveOrder.OrderType.SPECIFIED);
 
-        put(EXPORT_PRIMARY_SORT_FIELD, saveOrder.getSortCriteria().get(0).field.getName());
-        put(EXPORT_SECONDARY_SORT_FIELD, saveOrder.getSortCriteria().get(1).field.getName());
-        put(EXPORT_TERTIARY_SORT_FIELD, saveOrder.getSortCriteria().get(2).field.getName());
-        putBoolean(EXPORT_PRIMARY_SORT_DESCENDING, saveOrder.getSortCriteria().get(0).descending);
-        putBoolean(EXPORT_SECONDARY_SORT_DESCENDING, saveOrder.getSortCriteria().get(1).descending);
-        putBoolean(EXPORT_TERTIARY_SORT_DESCENDING, saveOrder.getSortCriteria().get(2).descending);
+        long saveOrderCount = saveOrder.getSortCriteria().stream().count();
```
Comment (koppor) [inline src/main/java/org/jabref/preferences/JabRefPreferences.java:2262]: Isn't `saveOrder.getSortCriteria().size()` working? That method returns a Java `List`, which should offer some kind of size reader.

**Review by koppor — CHANGES_REQUESTED** [review]
Tried out locally. Works well. 

Nitpick remaining ^^.

File: src/main/java/org/jabref/preferences/JabRefPreferences.java (hunk 2 above)
Comment (papatekken) [reply src/main/java/org/jabref/preferences/JabRefPreferences.java]: method replaced, thanks for the suggestion

File: CHANGELOG.md [hunk 3]
```diff
@@ -20,6 +20,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 
 - It is possible again to use "current table sort order" for the order of entries when saving. [#9869](https://github.com/JabRef/jabref/issues/9869)
 - Passwords can be stored in GNOME key ring. [#10274](https://github.com/JabRef/jabref/issues/10274)
+- we fixed a bug where exception was raised when saving less than 3 export save order in preference. [#10157](https://github.com/JabRef/jabref/issues/10157)
 - We fixed an issue were  groups based on an aux file could not be created due to an exception [#10350](https://github.com/JabRef/jabref/issues/10352)
```
Comment (Siedlerchr) [inline CHANGELOG.md:24]:
```suggestion
- We fixed an issue where  groups based on an aux file could not be created due to an exception [#10350](https://github.com/JabRef/jabref/issues/10352)
```

**Review by Siedlerchr — DISMISSED** [review]

**Comment by Siedlerchr** [comment]
Thanks for the update! looks good

**Review by calixtus — DISMISSED** [review]
Does not look very beautiful, but I guess it's the correct and there is no nicer way to do this. Therefor looks 'good' to me 😄

**Review by Siedlerchr — APPROVED** [review]

**Comment by papatekken** [comment]
> Does not look very beautiful, but I guess it's the correct and there is no nicer way to do this. Therefor looks 'good' to me 😄

I agree it does not look very nice. 
To do it in a better way it may need a bigger change with the data structure for the saving order preference, and the current preference export file  need change too.

**Pull request merged**